driver-reactive-streams/src/test/functional/com/mongodb/reactivestreams/client/syncadapter/SyncMongoCollection.java

```diff
@@ -0,0 +1,830 @@
+        wrapped.estimatedDocumentCount(options).subscribe(subscriber);
+        return requireNonNull(subscriber.first());
+    }
+
+    @Override
+    public <TResult> DistinctIterable<TResult> distinct(final String fieldName, final Class<TResult> resultClass) {
+        return new SyncDistinctIterable<>(wrapped.distinct(fieldName, resultClass));
+    }
+
+    @Override
+    public <TResult> DistinctIterable<TResult> distinct(final String fieldName, final Bson filter, final Class<TResult> resultClass) {
+        return new SyncDistinctIterable<>(wrapped.distinct(fieldName, filter, resultClass));
+    }
+
+    @Override
+    public <TResult> DistinctIterable<TResult> distinct(final ClientSession clientSession, final String fieldName,
+                                                        final Class<TResult> resultClass) {
+        return new SyncDistinctIterable<>(wrapped.distinct(unwrap(clientSession), fieldName, resultClass));
+    }
+
+    @Override
+    public <TResult> DistinctIterable<TResult> distinct(final ClientSession clientSession, final String fieldName, final Bson filter,
+                                                        final Class<TResult> resultClass) {
+        return new SyncDistinctIterable<>(wrapped.distinct(unwrap(clientSession), fieldName, filter, resultClass));
+    }
+
+    @Override
+    public FindIterable<T> find() {
+        return new SyncFindIterable<>(wrapped.find());
+    }
+
+    @Override
+    public <TResult> FindIterable<TResult> find(final Class<TResult> resultClass) {
+        return new SyncFindIterable<>(wrapped.find(resultClass));
+    }
+
+    @Override
+    public FindIterable<T> find(final Bson filter) {
+        return new SyncFindIterable<>(wrapped.find(filter));
+    }
+
+    @Override
+    public <TResult> FindIterable<TResult> find(final Bson filter, final Class<TResult> resultClass) {
+        return new SyncFindIterable<>(wrapped.find(filter, resultClass));
+    }
+
+    @Override
+    public FindIterable<T> find(final ClientSession clientSession) {
+        return new SyncFindIterable<>(wrapped.find(unwrap(clientSession)));
+    }
+
+    @Override
+    public <TResult> FindIterable<TResult> find(final ClientSession clientSession, final Class<TResult> resultClass) {
+        return new SyncFindIterable<>(wrapped.find(unwrap(clientSession), resultClass));
+    }
+
+    @Override
+    public FindIterable<T> find(final ClientSession clientSession, final Bson filter) {
+        return new SyncFindIterable<>(wrapped.find(unwrap(clientSession), filter));
+    }
+
+    @Override
+    public <TResult> FindIterable<TResult> find(final ClientSession clientSession, final Bson filter, final Class<TResult> resultClass) {
+        return new SyncFindIterable<>(wrapped.find(unwrap(clientSession), filter, resultClass));
+    }
+
+    @Override
+    public AggregateIterable<T> aggregate(final List<? extends Bson> pipeline) {
+        // TODO: mismatch between sync and async APIs
```

Comment:
How are these mismatched?

Reply:
In the sync API the return type is `AggregateIterable<Document>`